cdnsystem/daemon/cdn/manager.go

```diff
@@ -86,8 +90,27 @@ func (cm *Manager) TriggerCDN(ctx context.Context, task *types.SeedTask) (seedTa
 	cm.cdnLocker.Lock(task.TaskID, false)
 	defer cm.cdnLocker.UnLock(task.TaskID, false)
 	// first: detect Cache
-	fileMd5 := md5.New()
-	detectResult, err := cm.detector.detectCache(task, fileMd5)
+	var fileDigest hash.Hash
+	var digestType string
+	if !stringutils.IsBlank(task.RequestDigest) {
+		requestDigest := digestutils.Parse(task.RequestDigest)
+		digestType = requestDigest[0]
+		//requestDigest, _ := digest.Parse(task.RequestDigest)
+		//digestType = requestDigest.Algorithm().String()
+		switch digestType {
+		case digestutils.MD5:
+			fileDigest = md5.New()
+		case digestutils.SHA256:
+			fileDigest = sha256.New()
+		default:
+			digestType = digestutils.MD5
+			fileDigest = md5.New()
+		}
+	} else {
```

Comment:
why type is `MD5` when task.RequestDigest is empty.

Reply:
Ensure that the hash.Hash variable fileDigest is not nil when passed into the function detectCache